cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,880 @@
+
+        # copy nodes and make them TimeSeriesNodes
+        ts_cg = TimeSeriesCausalGraph()
+        for node in causal_graph.get_nodes():
+            meta = deepcopy(node.meta)
+            # get the variable name and lag from the node name
+            variable_name, lag = get_variable_name_and_lag(node.identifier)
+            node = TimeSeriesNode(variable_name=variable_name, time_lag=lag, meta=meta)
+            ts_cg.add_node(node)
+
+        # copy edges
+        for edge in causal_graph.get_edges():
+            source = ts_cg.get_node(edge.source)
+            destination = ts_cg.get_node(edge.destination)
+            ts_cg.add_edge(source, destination, meta=edge.meta)
+
+        ts_cg._sepsets = sepsets
+
+        return ts_cg
+
+    @staticmethod
+    def from_adjacency_matrix(
+        adjacency_matrix: numpy.ndarray,
+        node_names: Optional[List[Union[NodeLike, int]]] = None,
+    ) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from an adjacency matrix.
+        This is useful for converting an adjacency matrix into a time series causal graph.
+        """
+        graph = CausalGraph.from_adjacency_matrix(adjacency_matrix, node_names=node_names)
+        return TimeSeriesCausalGraph.from_causal_graph(graph)
+
+    @staticmethod
+    def from_adjacency_matrices(
+        adjacency_matrices: Dict[int, numpy.ndarray],
+        variable_names: Optional[List[Union[NodeLike, int]]] = None,
+    ) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from a dictionary of adjacency matrices. Keys are the time delta.
+        This is useful for converting a list of adjacency matrices into a time series causal graph.
+
+        For example, the adjacency matrix with time delta -1 is stored in adjacency_matrices[-1] as would correspond to X-1 -> X,
+        where X is the set of nodes.
+
+        Example:
+        >>> adjacency_matrices = {
+        ...     -2: numpy.array([[0, 0, 0], [1, 0, 0], [0, 0, 1]]),
+        ...     -1: numpy.array([[0, 1, 0], [1, 0, 0], [0, 0, 0]]),
+        ...     0: numpy.array([[0, 1, 1], [0, 0, 1], [0, 0, 0]]),
+        ... }
+
+        Let the nodes be X,Y,Z. The corresponding edges are:
+        >>> edges = [
+        ...     (Z-2, Y),
+        ...     (X-1, X),
+        ...     (Y-1, X),
+        ...     (Y-1, Y),
+        ...     (Z-1, Z),
+        ...     (X, Y),
+        ...     (X, Z),
+        ...     (Y, Z),
+        ... ]
+
+        :param adjacency_matrices: A dictionary of adjacency matrices. Keys are the time delta.
+        :param variable_names: A list of variable names. If not provided, the variable names are integers starting from 0.
+            Node names must correspond to the variable names and must not contain the lag.
+        :return: A time series causal graph.
+        """
+        assert isinstance(adjacency_matrices, dict)
+        # keys must be integers or str that can be converted to integers
```

Comment:
Why do node names need to be converted from strings to integers? This seems unnecessary to me. If it also should not contain lags, then does `len(variable_names)` have to be less than dim of adjacency matrix? Also swap between variable and node names in the docstring and is not clear what the difference is.

Reply:
I think this is just referring to the keys of the adjacency_matrices dict, so saying the keys which correspond to lags must be integers or integer like. Each adjacency matrix should be the same shape, len(variable_names)xlen(variable_names)